lib/v2/atcoder/contest.js

```diff
@@ -0,0 +1,64 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const status = ['action', 'upcoming', 'recent'];
+
+    const language = ctx.params.language ?? 'en';
+
+    let rated = ctx.params.rated ?? '0';
+    const category = ctx.params.category ?? '0';
+    const keyword = ctx.params.keyword ?? '';
+
+    rated = rated === 'active' ? 'action' : rated;
+    const isStatus = status.includes(rated);
+
+    const rootUrl = 'https://atcoder.jp';
+    const currentUrl = `${rootUrl}/contests${isStatus ? `?lang=${language}` : `/archive?lang=${language}&ratedType=${rated}&category=${category}${keyword ? `&keyword=${keyword}` : ''}`}`;
+
+    const response = await got({
+        method: 'get',
+        url: currentUrl,
+    });
+
+    const $ = cheerio.load(response.data);
+
+    let items = $(isStatus ? `#contest-table-${rated}` : '.row')
+        .find('tr')
+        .slice(1, ctx.query.limit ? parseInt(ctx.query.limit) : 20)
+        .toArray()
+        .map((item) => {
+            item = $(item).find('td a').eq(1);
+
+            return {
+                title: item.text(),
+                link: `${rootUrl}${item.attr('href')}`,
+            };
+        });
+
+    items = await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: item.link,
```

Comment:
Thank you very much for help on my issue.

I found that the item description is still Japan, so I suggest to add language param for item detail.
You can check with your Preview link: https://rsshub-682rmkf7u-diy.vercel.app/atcoder/contest/en/upcoming

```suggestion
                    url: `${item.link}?lang=${language}`,
```

Reply:
Thanks for the review 😊

Reply:
But why the output is still Japanese? Interesting, maybe a bug:

https://rsshub-1ei9fpduf-diy.vercel.app/atcoder/contest/en/upcoming